contracts/test/StakeTable.t.sol

```diff
@@ -0,0 +1,225 @@
+            BN254.G1Point(blsSigX, blsSigY) // sig
+        );
+    }
+
+    function setUp() public {
+        exampleTokenCreator = msg.sender;
+        vm.prank(exampleTokenCreator);
+        token = new ExampleToken(INITIAL_BALANCE);
+        tokenAddress = address(token);
+
+        LightClient.LightClientState memory genesis = LightClient.LightClientState({
+            viewNum: 0,
+            blockHeight: 0,
+            blockCommRoot: 0,
+            feeLedgerComm: 0,
+            stakeTableBlsKeyComm: 0,
+            stakeTableSchnorrKeyComm: 0,
+            stakeTableAmountComm: 0,
+            threshold: 0
+        });
+        lightClientContract = new LightClient(genesis,10);
+        lightClientAddress = address(lightClientContract);
+
+        stakeTable = new S(tokenAddress,lightClientAddress);
+    }
+
+    /// @dev Tests a correct key registation
+    function testRegisterHappyPath() external {
+        address msgSenderAddress = msg.sender;
+
+        (
+            BN254.G2Point memory blsVK,
+            EdOnBN254.EdOnBN254Point memory schnorrVK,
+            BN254.G1Point memory sig
+        ) = computeGroupElements(msgSenderAddress);
+
+        // Prepare for the token transfer
+        vm.prank(msg.sender);
+        token.approve(address(stakeTable), depositAmount);
+
+        vm.prank(msgSenderAddress);
+
+        // Balances before registration
+        assertEq(token.balanceOf(msgSenderAddress), INITIAL_BALANCE);
+
+        uint256 nativeAmount;
+        uint256 restakedAmount;
+        (nativeAmount, restakedAmount) = stakeTable.totalStake();
+        assertEq(nativeAmount, 0);
+        assertEq(restakedAmount, 0);
+
+        // Check event is emitted
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        IStakeTable.Node memory node;
+        node.account = msgSenderAddress;
+        node.balance = depositAmount;
+        node.stakeType = IStakeTable.StakeType.Native;
+        node.schnorrVK = schnorrVK;
+        node.registerEpoch = 1;
+
+        // The function IStakeTable._hashBlsKey is private but its code is small hence we copy it
+        // there.
+        emit Registered(
+            keccak256(abi.encode(blsVK.x0, blsVK.x1, blsVK.y0, blsVK.y1)),
+            node.registerEpoch,
+            node.stakeType,
+            node.balance
+        );
+
+        // Successful call to register
+        vm.prank(msgSenderAddress);
+        bool res = stakeTable.register(
+            blsVK, schnorrVK, depositAmount, IStakeTable.StakeType.Native, sig, validUntilEpoch
+        );
+        assertTrue(res);
+
+        // Balance after registration
+        assertEq(token.balanceOf(msgSenderAddress), INITIAL_BALANCE - depositAmount);
+        (nativeAmount, restakedAmount) = stakeTable.totalStake();
+        assertEq(nativeAmount, depositAmount);
+        assertEq(restakedAmount, 0);
+    }
+
+    function testInvalidBLSSig() external {
```

Comment:
we follow this [guide 2nd & 6th point](https://book.getfoundry.sh/tutorials/best-practices#general-test-guidance) for naming:

- `test[Fuzz]_revert[When|IF]_InvalidBLSSig`
- `test_Description_matches|succeeds`

Reply:
Done in db06877 and d4893b9f3cab0d05d0fea65242c693a77ca06aba.